### ISSUE TITLE
GRPO训练Qwen3-4B，在几十个Step后卡住

### ISSUE THREAD

**Issue by Mr-KenLee** [hiyouga/EasyR1#289]
问题如标题所示，配置如下：
```yaml
data:
  train_files: xxx
  val_files: xxx
  prompt_key: prompt
  answer_key: answer
  image_key: images
  max_prompt_length: 3072
  max_response_length: 1280
  rollout_batch_size: 32
  val_batch_size: 128
  format_prompt: null
  override_chat_template: "{%- if messages[0].role != 'system' -%}{{ '<|im_start|>system\\nYou are a helpful assistant!<|im_end|>\\n' -}}{%- endif -%}{%- for message in messages -%}{{ '<|im_start|>' ~ message.role ~ '\\n' ~ message.content ~ '<|im_end|>' ~ '\\n' -}}{%- endfor -%}{%- if add_generation_prompt -%}{{ '<|im_start|>assistant\\n' -}}{%- if enable_thinking is defined and not enable_thinking -%}{{ '<think>\\n\\n</think>\\n\\n' -}}{%- endif -%}{%- endif -%}"
  shuffle: true
  seed: 0
  max_pixels: 4194304
  min_pixels: 262144
  filter_overlong_prompts: false

algorithm:
  adv_estimator: grpo
  disable_kl: false
  use_kl_loss: true
  kl_penalty: low_var_kl
  kl_coef: 1.0e-2

worker:
  actor:
    global_batch_size: 32
    micro_batch_size_per_device_for_update: 2
    micro_batch_size_per_device_for_experience: 8
    max_grad_norm: 1.0
    padding_free: true
    ulysses_sequence_parallel_size: 1
    model:
      model_path: xxx
      enable_gradient_checkpointing: true
      trust_remote_code: false
      freeze_vision_tower: false
    optim:
      lr: 2.0e-6
      weight_decay: 1.0e-3
      strategy: adamw
      lr_warmup_ratio: 0.1
    fsdp:
      enable_full_shard: true
      enable_cpu_offload: false
      enable_rank0_init: true
    offload:
      offload_params: true
      offload_optimizer: true
  rollout:
    n: 4
    temperature: 1.0
    top_p: 0.99
    gpu_memory_utilization: 0.6
    enforce_eager: false
    enable_chunked_prefill: false
    tensor_parallel_size: 8
    max_num_batched_tokens: 8192
    limit_images: 0
    dtype: "bf16"
    val_override_config:
      temperature: 0.0
      n: 1  
  ref:
    fsdp:
      enable_full_shard: true
      enable_cpu_offload: true
      enable_rank0_init: true
    offload:
      offload_params: false
  reward:
    reward_type: batch
    reward_function: ./MyConfig/reward.py:compute_score

trainer:
  total_epochs: 3
  max_steps: null
  project_name: easy_r1
  experiment_name: grpo_test
  logger: ["console"]
  nnodes: 1
  n_gpus_per_node: 8
  val_freq: 1000
  val_before_train: true
  val_only: false
  val_generations_to_log: 1
  save_freq: 1000
  save_limit: 3
  save_checkpoint_path: null
  load_checkpoint_path: null
```

环境配置为：
vllm==0.8.5.post1, pytorch==2.6.0, transformers==4.51.1, accelerate==1.6.0

**Comment by Mr-KenLee** [comment]
显存的占用和利用率一直有正常的变动，但是log卡住不动了，不知道是不是在正常训练。
```shell
(WorkerDict pid=141364) Before state_dict() in sharding manager: 10.07 GB / 39.59 GB.       scripts   verl
(WorkerDict pid=141364) After state_dict() in sharding manager: 12.42 GB / 39.59 GB.ts.txt  setup.py  verl.egg-info
(WorkerDict pid=141364) After sync model weights in sharding manager: 13.45 GB / 39.59 GB.ab-docker-4968-1-6b7647bcc4-v9hpw 10:36:16
(WorkerDict pid=141364) After del state_dict and empty_cache in sharding manager: 30.85 GB / 39.59 GB.
Processed prompts:   0%|          | 0/128 [00:00<?, ?it/s, est. speed input: 0.00 toks/s, output: 0.00 toks/s]
Processed prompts:   3%|▎         | 4/128 [00:13<07:03,  3.41s/it, est. speed input: 394.90 toks/s, output: 95.79 toks/s]
Processed prompts:   6%|▋         | 8/128 [00:16<03:46,  1.89s/it, est. speed input: 638.03 toks/s, output: 183.14 toks/s]
Processed prompts:   9%|▉         | 12/128 [00:17<02:01,  1.05s/it, est. speed input: 946.82 toks/s, output: 271.62 toks/s]
Processed prompts:  12%|█▎        | 16/128 [00:17<01:19,  1.41it/s, est. speed input: 1208.63 toks/s, output: 348.09 toks/s]
Processed prompts:  16%|█▌        | 20/128 [00:18<00:56,  1.90it/s, est. speed input: 1444.12 toks/s, output: 432.18 toks/s]
Processed prompts:  19%|█▉        | 24/128 [00:19<00:39,  2.65it/s, est. speed input: 1701.83 toks/s, output: 525.20 toks/s]
Processed prompts:  22%|██▏       | 28/128 [00:19<00:29,  3.39it/s, est. speed input: 1936.54 toks/s, output: 624.73 toks/s]
Processed prompts:  25%|██▌       | 32/128 [00:19<00:22,  4.25it/s, est. speed input: 2168.14 toks/s, output: 715.74 toks/s]
Processed prompts:  28%|██▊       | 36/128 [00:20<00:19,  4.69it/s, est. speed input: 2359.11 toks/s, output: 801.15 toks/s]
Processed prompts:  31%|███▏      | 40/128 [00:23<00:33,  2.66it/s, est. speed input: 2289.91 toks/s, output: 792.36 toks/s]
Processed prompts:  34%|███▍      | 44/128 [00:24<00:30,  2.78it/s, est. speed input: 2386.45 toks/s, output: 869.32 toks/s]
Processed prompts:  38%|███▊      | 48/128 [00:25<00:26,  3.02it/s, est. speed input: 2554.57 toks/s, output: 943.75 toks/s]
Processed prompts:  41%|████      | 52/128 [00:26<00:19,  3.87it/s, est. speed input: 2724.55 toks/s, output: 1039.01 toks/s]
Processed prompts:  44%|████▍     | 56/128 [00:26<00:16,  4.42it/s, est. speed input: 2862.02 toks/s, output: 1101.04 toks/s]
Processed prompts:  47%|████▋     | 60/128 [00:27<00:12,  5.55it/s, est. speed input: 3029.77 toks/s, output: 1179.47 toks/s]
Processed prompts:  50%|█████     | 64/128 [00:27<00:08,  7.12it/s, est. speed input: 3203.55 toks/s, output: 1246.88 toks/s]
Processed prompts:  53%|█████▎    | 68/128 [00:27<00:06,  8.89it/s, est. speed input: 3403.50 toks/s, output: 1341.24 toks/s]
Processed prompts:  56%|█████▋    | 72/128 [00:27<00:05, 11.07it/s, est. speed input: 3577.76 toks/s, output: 1441.77 toks/s]
Processed prompts:  62%|██████▎   | 80/128 [00:28<00:03, 13.65it/s, est. speed input: 3904.22 toks/s, output: 1665.27 toks/s]
Processed prompts:  66%|██████▌   | 84/128 [00:28<00:04, 10.54it/s, est. speed input: 4002.99 toks/s, output: 1736.94 toks/s]
(WorkerDict pid=141364) Before vllm offload in sharding manager: 27.43 GB / 39.59 GB.   
Processed prompts: 100%|██████████| 128/128 [00:29<00:00,  4.31it/s, est. speed input: 6036.34 toks/s, output: 3047.74 toks/s]
(WorkerDict pid=141364) After vllm offload in sharding manager: 5.83 GB / 39.59 GB.     
(pid=131906) Epoch 0:  33%|███▎      | 1.00/3.00 [1:03:03<2:06:07, 3.78ks/it]           
(pid=131906) Running step 1:   6%|▌         | 53.0/912 [1:03:03<24:49:37, 104s/it]      
                                                                                        
                                                                                        
(pid=141364) Train mini-batches 2:   0%|          | 0.00/1.00 [00:00<?, ?it/s]0kit/s]]t]
(pid=141364) Update policy 3: 100%|██████████| 8.00/8.00 [00:00<00:00, 11.2kit/s]]
```

**Comment by Mr-KenLee** [comment]
启动训练的脚本如下：
```shell
#!/bin/bash

set -x

export VLLM_USE_V1=0
export TORCH_CUDA_ARCH_LIST="8.0;8.6"
export PYTHONUNBUFFERED=1

nohup python3 -m verl.trainer.main \
    config=MyConfig/config.yaml > output.log 2>&1 &
```

**Comment by ssmisya** [comment]
+1, 我也遇到了类似的问题，Qwen 2.5 VL在训练几个小时后主进程的log卡住，但是ray worker的log是在继续输出的，且wandb和ckpt的输出也停止了，想知道如何排查原因

**Comment by hiyouga** [comment]
只要 ray log 正常输出就说明训练在继续，建议以 ray log 和 wandb 为准

**Comment by diegorlady** [comment]
Reopening this issue and adding `qwen3-related` label as requested.
